tests/mock_code/test_ignore_paths.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""
+Test that ignoring paths works as expected.
+"""
+import os
+from pathlib import Path
+import pytest
+
+from aiida_testing.mock_code._cli import copy_files
+
+OUTPUT_PATHS = (
+    Path('file1.txt'),
+    Path('file2.txt'),
+    Path('_aiidasubmit.sh'),
+    Path('my/subfolder/file3.txt'),
+    Path('my/subfolder/file4.txt'),
+)
+
+
+@pytest.fixture
+def run_directory(tmp_path_factory):
+    """
+    Prepare mock directory structure for run directory of code.
+    """
+    tmp_path = tmp_path_factory.mktemp('output')
+    for path in OUTPUT_PATHS:
+        if os.path.dirname(path):
+            os.makedirs(tmp_path / os.path.dirname(path), exist_ok=True)
+        with open(tmp_path / path, 'w') as handle:
+            handle.write("Test content")
+
+    yield tmp_path
+
+
+def test_ignore_paths(run_directory, tmp_path_factory):  # pylint: disable=redefined-outer-name
+    """Test that ignore_paths works as expected."""
+    storage_directory = tmp_path_factory.mktemp('storage')
+
+    # ignore everything
+    copy_files(
+        src_dir=run_directory, dest_dir=storage_directory, ignore_files=(), ignore_paths=('*', )
+    )
+    assert not (storage_directory / '_aiidasubmit.sh').is_file()
+
+    # ignore subfolder
+    copy_files(
+        src_dir=run_directory, dest_dir=storage_directory, ignore_files=(), ignore_paths=('my/', )
+    )
+    copy_files(
+        src_dir=run_directory,
+        dest_dir=storage_directory,
+        ignore_files=(),
+        ignore_paths=('my/subfolder', )
+    )
+    assert (storage_directory / '_aiidasubmit.sh').is_file()
+    assert (storage_directory / 'file2.txt').is_file()
+    assert not (storage_directory / 'my' / 'subfolder' / 'file3.txt').is_file()
+
+    # ignore only specific file (from subfolder)
+    copy_files(
```

Comment:
I think we should add a copy of this specific test where `ignore_paths` is used instead of `ignore_files`.

Reply:
Thanks for the comment - this made me think that `rglob` is probably not the best solution here for `ignore_paths` (and I went back to `glob`).

I've added another test for `ignore_files` now; please have a look

Reply:
That means once we remove `ignore_files` one would just have to use `**/filename` to ignore everything of a given name, right? I think that's fine (more explicit).